rcode-reader/src/main/java/eu/rssw/pct/elements/MethodElement.java

```diff
@@ -22,37 +22,47 @@
   protected static final int OVERLOADED_METHOD = 256;
   protected static final int STATIC_METHOD = 512;
 
-  protected int flags;
-  protected int returnType;
-
-  protected String typeName;
-  protected int extent;
-  protected IParameter[] parameters;
-
-  public MethodElement(String name, Set<AccessType> accessType, byte[] segment, int currentPos, int textAreaOffset,
-      ByteOrder order) {
+  private final int flags;
+  private final int returnType;
+  private final String returnTypeName;
+  private final int extent;
+  private final IParameter[] parameters;
+
+  public MethodElement(String name, Set<AccessType> accessType, int flags, int returnType, String returnTypeName,
+      int extent, IParameter[] parameters) {
     super(name, accessType);
+    this.flags = flags;
+    this.returnType = returnType;
+    this.returnTypeName = returnTypeName;
+    this.extent = extent;
+    this.parameters = parameters;
+  }
 
-    this.flags = ByteBuffer.wrap(segment, currentPos, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
-    this.returnType = ByteBuffer.wrap(segment, currentPos + 2, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
+  public static MethodElement fromDebugSegment(String name, Set<AccessType> accessType, byte[] segment, int currentPos, int textAreaOffset,
+      ByteOrder order) {
+    int flags = ByteBuffer.wrap(segment, currentPos, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
+    int returnType = ByteBuffer.wrap(segment, currentPos + 2, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
     int paramCount = ByteBuffer.wrap(segment, currentPos + 4, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
-    this.extent = ByteBuffer.wrap(segment, currentPos + 8, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
+    int extent = ByteBuffer.wrap(segment, currentPos + 8, Short.BYTES).order(ByteOrder.LITTLE_ENDIAN).getShort();
 
     int nameOffset = ByteBuffer.wrap(segment, currentPos + 12, Integer.BYTES).order(ByteOrder.LITTLE_ENDIAN).getInt();
-    this.name = nameOffset == 0 ? "" : RCodeInfo.readNullTerminatedString(segment, textAreaOffset + nameOffset);
+    String name2 = nameOffset == 0 ? name : RCodeInfo.readNullTerminatedString(segment, textAreaOffset + nameOffset);
 
     int typeNameOffset = ByteBuffer.wrap(segment, currentPos + 16, Integer.BYTES).order(
         ByteOrder.LITTLE_ENDIAN).getInt();
-    this.typeName = typeNameOffset == 0 ? ""
+    String typeName = typeNameOffset == 0 ? ""
         : RCodeInfo.readNullTerminatedString(segment, textAreaOffset + typeNameOffset);
 
+    System.out.println("meth " + name + " -- " + name2 + " -- " + typeName + " -- " + returnType);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS106)